asdf/resolver.py

```diff
@@ -1,16 +1,28 @@
-import sys
 import warnings
 
 from . import _resolver
 from .exceptions import AsdfDeprecationWarning
 
-warnings.warn(
-    "asdf.resolver is deprecated "
-    "Please see Resources "
-    "https://asdf.readthedocs.io/en/stable/asdf/extending/resources.html",
-    AsdfDeprecationWarning,
-)
 
-# overwrite the hidden module __file__ so pytest doesn't throw an ImportPathMismatchError
-_resolver.__file__ = __file__
-sys.modules[__name__] = _resolver
+def _warn():
+    warnings.warn(
+        "asdf.resolver is deprecated "
+        "Please see Resources "
+        "https://asdf.readthedocs.io/en/stable/asdf/extending/resources.html",
+        AsdfDeprecationWarning,
+    )
+
+
+_warn()
+
+
+def __getattr__(name):
+    attr = getattr(_resolver, name)
+    _warn()
+    if hasattr(attr, "__module__"):
+        attr.__module__ = __name__
+    return attr
```

Comment:
Can you add a test to `test_deprecated.py` to actually test the functionality of this method?